### PR TITLE
CD-102 dropdowns

### PR DESCRIPTION
**Pull request by left23**
See [CD-102](https://humanitarian.atlassian.net/browse/CD-102) for notes on this.

This PR prevents having two menu dropdowns open at once (see https://ochabasic.dev.ahconu.org) and it leverages bootstrap dropdown plugin behaviour to close dropdowns when clicking outside them.

It does not fix the instance where the language switcher remains open when a menu dropdown is opened. This is relevant on desktop only, because on mobile, the language switcher when opened, hides the menu toggle.

It also replaces the pseudo elements for the chevrons with svg icons. Since I was using a menu preprocess function anyway, it seemed an opportunity to improve.

### Update
I have removed the 3rd level dropdown menu support, which means we can use bootstrap dropdown plugin out-of-the-box for the main menu dropdowns by relying on the data-toggle attributes. 

Due to the mobile menu toggle being a dropdown and therefore adding an extra level, I needed to make adjustments to display the mobile menu and keep its dropdowns intact.

The only custom JS remaining deals with the mobile menu toggle - toggling the class "open" and setting display properties in CSS.

In order to make sure the menu closes when other dropdowns are open (search, ocha services, language switcher) I check for visibility of the mobile menu and remove the "open" class. This emulates the behaviour from Bootstrap that we see on desktop. Result should be that only one element is open at once.

Note. matchMedia is not supported in IE9 so I might do well to use another method to check viewport width.

### CODE REVIEW

**Comment by left23** [comment]
@rupl I'm still of two minds about this. Let me know what you think when you get a chance to read and review. I'm in no rush to move on this. It can wait for further discussion and others opinions if you wish.

**Comment by rupl** [comment]
I agree it's a little wild but I appreciate the commenting you added. At least when we revisit this we have somewhat of a clue why/how it's working. I think if it solves the problem it's worth merging.

To me it's strange that three drop-downs can be open and it feels broken to me when I cause it to happen.

**Comment by left23** [comment]
@rupl Please have another look when you have a chance. This has changed since we decided not to cater for more than 2 levels for dropdown menu. It is simpler now only because the custom JS is limited to the mobile menu behaviour, whereas the desktop menu uses all Bootstrap dropdown.js

File: js/custom-menu.js [hunk 1]
```diff
@@ -6,11 +6,32 @@
 (function ($, Drupal) {
   Drupal.behaviors.cdMenu = {
     attach: function (context, settings) {
-      $('.cd-dropdown .expanded > a').on('click', function (e) {
-        $(this).parent().toggleClass('open');
-        e.stopPropagation();
-        e.preventDefault();
-      });
+
+      // Media query event handler.
```
Comment (rupl) [inline js/custom-menu.js]: Can you describe the purpose of the handler in the comment? Doesn't need to be extensive but it's the kind of explainer that will probably be useful when we revisit in the future.

Comment (left23) [reply js/custom-menu.js]: For sure, added a few lines.

**Pull request merged**